Issue while using new PGI compiler via NVIDIA HPC SDK

PGI compilers are no longer available and they are now part of [NVIDIA HPC SDK](https://developer.nvidia.com/hpc-sdk). Compiler names are now changed (nvc, nvc++, nvfortran) and hence CMake no longer detects them:

```
$ cmake ..
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- Check for working C compiler: /gpfs/bbp.cscs.ch/home/kumbhar/tmp/soft/install/linux-rhel7-x86_64/gcc-8.3.0/nvhpc-20.9-rrs2t4/Linux_x86_64/20.9/compilers/bin/nvc
-- Check for working C compiler: /gpfs/bbp.cscs.ch/home/kumbhar/tmp/soft/install/linux-rhel7-x86_64/gcc-8.3.0/nvhpc-20.9-rrs2t4/Linux_x86_64/20.9/compilers/bin/nvc -- broken
CMake Error at /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/cmake-3.15.3-cccsbe7omc/share/cmake-3.15/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "/gpfs/bbp.cscs.ch/home/kumbhar/tmp/soft/install/linux-rhel7-x86_64/gcc-8.3.0/nvhpc-20.9-rrs2t4/Linux_x86_64/20.9/compilers/bin/nvc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /gpfs/bbp.cscs.ch/home/kumbhar/tmp/CoreNeuron/build/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/gmake cmTC_3e5d4/fast && /usr/bin/gmake -f CMakeFiles/cmTC_3e5d4.dir/build.make CMakeFiles/cmTC_3e5d4.dir/build
    gmake[1]: Entering directory `/gpfs/bbp.cscs.ch/home/kumbhar/tmp/CoreNeuron/build/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_3e5d4.dir/testCCompiler.c.o
    /gpfs/bbp.cscs.ch/home/kumbhar/tmp/soft/install/linux-rhel7-x86_64/gcc-8.3.0/nvhpc-20.9-rrs2t4/Linux_x86_64/20.9/compilers/bin/nvc    -o CMakeFiles/cmTC_3e5d4.dir/testCCompiler.c.o   -c /gpfs/bbp.cscs.ch/home/kumbhar/tmp/CoreNeuron/build/CMakeFiles/CMakeTmp/testCCompiler.c
    nvc-Error-Please run makelocalrc to complete your installation
    gmake[1]: *** [CMakeFiles/cmTC_3e5d4.dir/testCCompiler.c.o] Error 1
    gmake[1]: Leaving directory `/gpfs/bbp.cscs.ch/home/kumbhar/tmp/CoreNeuron/build/CMakeFiles/CMakeTmp'
    gmake: *** [cmTC_3e5d4/fast] Error 2

```

--- a/coreneuron/CMakeLists.txt
+++ b/coreneuron/CMakeLists.txt
@@ -89,10 +89,9 @@ if(CORENRN_ENABLE_GPU)
   # compile cuda files for multiple architecture
   cuda_add_library(
     "cudacoreneuron" ${CORENEURON_CUDA_FILES}
-    OPTIONS -arch=sm_30
-            -gencode=arch=compute_30,code=sm_30
-            -gencode=arch=compute_50,code=sm_50
-            -gencode=arch=compute_52,code=sm_52
+    OPTIONS -arch=sm_60
+            -gencode=arch=compute_60,code=sm_60
+            -gencode=arch=compute_72,code=sm_72
             -gencode=arch=compute_52,code=compute_52
             -Xcompiler
             -fPIC)

--- a/coreneuron/permute/cellorder.cpp
+++ b/coreneuron/permute/cellorder.cpp
@@ -463,9 +463,13 @@ static void triang_interleaved2(NrnThread* nt, int icore, int ncycle, int* strid
 #if !defined(_OPENACC)
     int ii = i;
 #endif
+
+    // execute until all tree depths are executed
+    bool has_subtrees_to_compute = true;
+
 // clang-format off
     #pragma acc loop seq
-    for (;;) {  // ncycle loop
+    for (; has_subtrees_to_compute; ) {  // ncycle loop
 #if !defined(_OPENACC)
         // serial test, gpu does this in parallel
         for (int icore = 0; icore < warpsize; ++icore) {
@@ -483,8 +487,11 @@ static void triang_interleaved2(NrnThread* nt, int icore, int ncycle, int* strid
 #if !defined(_OPENACC)
         }
 #endif
+        // if finished with all tree depths then ready to break
+        // (note that break is not allowed in OpenACC)
         if (icycle == 0) {
-            break;
+            has_subtrees_to_compute = false;
+            continue;
         }
         --icycle;
         istride = stride[icycle];

